applications/MeshMovingApplication/custom_utilities/explicit_mesh_moving_utilities.h

```diff
@@ -0,0 +1,255 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main author:     Ruben Zorrilla
+//
+
+#if !defined(KRATOS_EXPLICIT_MESH_MOVING_UTILITIES_H_INCLUDED )
+#define  KRATOS_EXPLICIT_MESH_MOVING_UTILITIES_H_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "spatial_containers/spatial_search.h"
+#include "spatial_containers/bins_dynamic_objects.h"
+#include "spatial_containers/configures/node_configure.h"
+
+namespace Kratos
+{
+  ///@addtogroup FluidDynamicsApplication
+  ///@{
+
+  ///@name Kratos Globals
+  ///@{
+
+  ///@}
+  ///@name Type Definitions
+  ///@{
+
+  ///@}
+  ///@name  Enum's
+  ///@{
+
+  ///@}
+  ///@name  Functions
+  ///@{
+
+  ///@}
+  ///@name Kratos Classes
+  ///@{
+
+  /// Utility to initialize the historical data in moving boundary problems
+  /** This utility is based on the Fixed Mesh - Arbitrary Lagrangian Eulerian
+   * (FM-ALE) method but solving the mesh problem in an explicit manner. Thus,
+   * a virtual mesh is set. This virtual mesh is moved according to the
+   * structure movement. The virtual mesh movement, is computed in an explicit 
```

Comment:
does this work only for a structure?
From what I understood I would say that this also works for other cases...?

In this case I would not use the name "structure" here.

Reply:
Well, I cannot find an application in where you use this method for anything which is not to embed a structure... In any case I would change "structure" by "embedded object".